src/application/services/crux-asset-translator.ts

```diff
@@ -0,0 +1,103 @@
+import { IAddress, IAddressMapping } from "../../core/entities/crux-user";
+import { errors } from "../../packages";
+import { IGlobalAssetList } from "../../packages/configuration-service";
+import { getLogger } from "../../packages/logger";
+const log = getLogger(__filename);
+
+export interface IClientAssetMapping {
+    [currencySymbol: string]: string;
+}
+
+export interface IGlobalAsset {
+    assetId: string;
+    symbol: string;
+    name: string;
+    assetType: string|null;
+    decimals: number|null;
+    assetIdentifierName: string|null;
+    assetIdentifierValue: number|string|null;
+    parentAssetId: string|null;
+}
+
+export interface IResolvedClientAssetMap {
+    [currencySymbol: string]: IGlobalAsset;
+}
+
+export interface IReverseClientAssetMapping {
+    [assetId: string]: string;
+}
+
+export interface IPutAddressMapSuccess {
+    [currency: string]: IAddress;
+}
+
+export interface IPutAddressMapFailures {
+    [currency: string]: string;
+}
+
+export class CruxAssetTranslator {
+    private _assetMapping: IClientAssetMapping;
+    private _reverseAssetMap: IReverseClientAssetMapping;
+    constructor(assetMapping: IClientAssetMapping) {
+        this._assetMapping = assetMapping;
+        this._reverseAssetMap = {};
+        for (let walletCurrencySymbol of Object.keys(this._assetMapping)) {
+            walletCurrencySymbol = walletCurrencySymbol.toLowerCase();
+            this._reverseAssetMap[this._assetMapping[walletCurrencySymbol]] = walletCurrencySymbol;
+        }
+    }
+    get assetMapping() {
+        return this._assetMapping;
+    }
+    set assetMapping(assetMapping: IClientAssetMapping) {
+        this._assetMapping = assetMapping;
+    }
+    public symbolToAssetId(currencySymbol: string): string {
+        currencySymbol = currencySymbol.toLowerCase();
+        return this._assetMapping[currencySymbol];
+    }
+    public assetIdToSymbol(assetId: string): string {
+        return this._reverseAssetMap[assetId];
+    }
+
+    public assetIdAssetMapToSymbolAssetMap(assetIdAssetMap: IGlobalAssetList): IResolvedClientAssetMap {
+        const symbolAssetMap: IResolvedClientAssetMap = {};
+        const currencySymbol = undefined;
+        assetIdAssetMap.forEach((asset) => {
```

Comment:
I am sure map object does not support `forEach` functionality.

Reply:
It's actually list renamed variable to assetIdAssetList